rancher2/config.go

```diff
@@ -1111,16 +1139,57 @@ func (c *Config) UpdateClusterByID(cluster *managementClient.Cluster, update map
 }
 
 func (c *Config) isClusterActive(id string) (bool, *managementClient.Cluster, error) {
-	clus, err := c.GetClusterByID(id)
+	obj, err := c.GetClusterByID(id)
 	if err != nil {
 		return false, nil, err
 	}
 
-	if clus.State == "active" {
-		return true, clus, nil
+	for i := range obj.Conditions {
+		if obj.Conditions[i].Type == clusterActiveCondition {
+			if obj.Conditions[i].Status == "True" {
+				return true, obj, nil
+			}
+			return false, obj, nil
+		}
+	}
+
+	return false, obj, nil
+}
+
+func (c *Config) isClusterMonitoringEnabled(id string) (bool, *managementClient.Cluster, error) {
+	obj, err := c.GetClusterByID(id)
+	if err != nil {
+		return false, nil, err
+	}
+
+	for i := range obj.Conditions {
+		if obj.Conditions[i].Type == clusterMonitoringEnabledCondition {
+			if obj.Conditions[i].Status == "True" {
+				return true, obj, nil
+			}
+			return false, obj, nil
+		}
+	}
+
+	return false, obj, nil
+}
+
+func (c *Config) isClusterAlertingEnabled(id string) (bool, *managementClient.Cluster, error) {
```

Comment:
Same as with `isClusterMonitoringEnabled`

Reply:
This function is checking if cluster condition, `clusterAlertingEnabledCondition   = "AlertingEnabled"` is true

Reply:
Updated function name to `isClusterAlertingEnabledCondition`